modules/atomic-widgets/prop-types/prop-type.php

```diff
@@ -33,11 +40,53 @@ public function get_default() {
 		return $this->default;
 	}
 
+	public function add_meta( array $meta ): self {
+		$is_tuple = 2 === count( $meta );
+
+		if ( ! $is_tuple ) {
+			return $this;
+		}
+
+		[ $key, $value ] = $meta;
+
+		if ( ! is_string( $key ) ) {
+			return $this;
+		}
+
+		$this->meta[ $key ] = $value;
+
+		return $this;
+	}
+
+	public function get_meta( string $key ) {
+		return $this->meta[ $key ] ?? null;
+	}
+
+	public function additional_type( Prop_Type $type ): self {
+		$this->additional_types[] = $type;
+
+		return $this;
+	}
+
+	public function get_additional_types(): array {
+		return $this->additional_types;
+	}
+
 	public function jsonSerialize(): array {
+		$additional_types = array_map( function ( $type ) {
+			return [
+				'key' => $type::get_key(),
+				'settings' => (object) $type->settings,
+			];
+		}, $this->additional_types );
+
```

Comment:
```suggestion
	public function jsonSerialize( $is_additional = false ): array {
	
	   $additional_types = $this->additional_types->map( fn( $type ) => $type->jsonSerialize( true ) );
	   
           $is_additional ? [
				'key' => $type::get_key(),
				'settings' => (object) $type->settings,
			] : [
			'type' => [
				'key' => static::get_key(),
				'default' => $this->default,
				'settings' => (object) $this->settings,
			],
			'additional_types' => $additional_types,
		];
        }
```

maybe something like this

Reply:
IDK, it's weird
@Nevoss ?